File: data/src/main/java/com/somename/data/repository/datasource/MoviesAPIDataSource.java
package com.somename.data.repository.datasource;

import com.somename.data.network.LocalApi;
import com.somename.data.сontent.RootEntity;
import com.somename.data.сontent.RootProductEntity;
import com.somename.data.сontent.RootTopicsEntity;

import io.reactivex.Observable;

public class MoviesAPIDataSource implements DataSource {

    private final LocalApi mLocalApi;

    public MoviesAPIDataSource(LocalApi localApi) {
        mLocalApi = localApi;
    }


    @Override
    public Observable<RootEntity> getTechCategoryEntity() {
        return mLocalApi.getTechCategory();
    }

    @Override
    public Observable<RootTopicsEntity> getTrendingTopicsEntity() {
        return mLocalApi.getTrendingTopics();
    }

    @Override
    public Observable<RootEntity> getChoosenTopicEntity(int topicId) {
        return mLocalApi.getChosenTopic(topicId);
    }

    @Override
    public Observable<RootProductEntity> getProductEntity(String id) {
        return mLocalApi.getProduct(id);
    }
}

File: data/src/main/java/com/somename/data/repository/datasource/MoviesAPIDataSource.kt
package com.somename.data.repository.datasource

import com.somename.data.network.LocalApi
import com.somename.data.сontent.RootEntity
import com.somename.data.сontent.RootProductEntity
import com.somename.data.сontent.RootTopicsEntity

import io.reactivex.Observable

class MoviesAPIDataSource(private val mLocalApi: LocalApi) : DataSource {


    override val techCategoryEntity: Observable<RootEntity>
        get() = mLocalApi.techCategory

    override val trendingTopicsEntity: Observable<RootTopicsEntity>
        get() = mLocalApi.trendingTopics

    override fun getChoosenTopicEntity(topicId: Int): Observable<RootEntity> {
        return mLocalApi.getChosenTopic(topicId)
    }

    override fun getProductEntity(id: String): Observable<RootProductEntity> {
        return mLocalApi.getProduct(id)
    }
}
